### PR TITLE
Fix use of docker-machine for integration test

### PR DESCRIPTION
**Pull request by parente**
Use docker links, not public IPs

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -164,12 +162,14 @@ $(DASHBOARD_SERVER) -d \
 @$(MAKE) test-container \
 	CMD=$(CMD) \
 	SERVER_NAME=$(IT_SERVER_NAME) \
-	DOCKER_OPTIONS="-e APP_URL=http://$(IT_IP):$(HTTP_PORT) \
-		-e KERNEL_GATEWAY_URL=http://$(IT_IP):$(IT_KG_PORT) \
+	DOCKER_OPTIONS="-e APP_URL=http://$(DASHBOARD_CONTAINER_NAME):$(HTTP_PORT) \
+		-e KERNEL_GATEWAY_URL=http://$(KG_CONTAINER_NAME):$(IT_KG_PORT) \
```
Comment (jhpedemonte) [inline Makefile]: `IT_KG_PORT` is still referenced in 2 places, but not defined. In other parts of the Makefile, we explicitly use `8888` with `KG_CONTAINER_NAME`. Probably best to pull that out into a `KG_PORT` var.

**Comment by jhpedemonte** [comment]
Auth tests still fail for me, on 2 machines running docker beta and docker-machine. This patch makes all integration tests pass:

```diff
diff --git a/Makefile b/Makefile
index a343e49..e91c118 100644
--- a/Makefile
+++ b/Makefile
@@ -13,6 +13,7 @@ KG_IMAGE_NAME:=jupyter-incubator/kernel-gateway-extras
 KG_CONTAINER_NAME:=kernel-gateway
 HTTP_PORT?=3000
 HTTPS_PORT?=3001
+KG_PORT?=8888
 TEST_CONTAINER_NAME:=$(DASHBOARD_CONTAINER_NAME)-test
 TEST_IMAGE_NAME:=jupyter-incubator/$(TEST_CONTAINER_NAME)
 
@@ -114,7 +115,7 @@ kernel-gateway-container:
        $(RUN_KERNEL_GATEWAY)
 
 # targets for running nodejs app and kernel gateway containers
-demo-container: KERNEL_GATEWAY_URL?=http://$(KG_CONTAINER_NAME):8888
+demo-container: KERNEL_GATEWAY_URL?=http://$(KG_CONTAINER_NAME):$(KG_PORT)
 demo-container: | build kernel-gateway-container ## Run dashboard server in a docker container
        @echo '-- Starting dashboard server container'
        $(DASHBOARD_SERVER) -it --rm \
@@ -150,7 +151,7 @@ define RUN_INTEGRATION_TEST
 $(RUN_KERNEL_GATEWAY)
 @echo '-- Starting dashboard server container'
 $(DASHBOARD_SERVER) -d \
-       -e KERNEL_GATEWAY_URL=http://$(KG_CONTAINER_NAME):8888 \
+       -e KERNEL_GATEWAY_URL=http://$(KG_CONTAINER_NAME):$(KG_PORT) \
        -e KG_AUTH_TOKEN=$(KG_AUTH_TOKEN) \
        -e AUTH_TOKEN=$(AUTH_TOKEN) \
        -e USERNAME=$(USERNAME) \
@@ -163,7 +164,7 @@ $(DASHBOARD_SERVER) -d \
        CMD=$(CMD) \
        SERVER_NAME=$(IT_SERVER_NAME) \
        DOCKER_OPTIONS="-e APP_URL=http://$(DASHBOARD_CONTAINER_NAME):$(HTTP_PORT) \
-               -e KERNEL_GATEWAY_URL=http://$(KG_CONTAINER_NAME):$(IT_KG_PORT) \
+               -e KERNEL_GATEWAY_URL=http://$(KG_CONTAINER_NAME):$(KG_PORT) \
                -e KG_AUTH_TOKEN=$(KG_AUTH_TOKEN) \
                -e AUTH_TOKEN=$(AUTH_TOKEN) \
                -e TEST_USERNAME=$(USERNAME) \
@@ -201,7 +202,7 @@ install-test: | kill build ## Run basic integration tests after npm install in a
        @docker build --rm -f Dockerfile.installed -t $(INSTALLED_DASHBOARD_IMAGE_NAME) .
        @echo '-- Running dashboard server...'
        $(DASHBOARD_SERVER) -d \
-               -e KERNEL_GATEWAY_URL=http://$(KG_CONTAINER_NAME):8888 \
+               -e KERNEL_GATEWAY_URL=http://$(KG_CONTAINER_NAME):$(KG_PORT) \
                --link $(KG_CONTAINER_NAME):$(KG_CONTAINER_NAME) \
                $(INSTALLED_DASHBOARD_IMAGE_NAME)
        @echo '-- Waiting 10 seconds for server to start...'
@@ -210,7 +211,7 @@ install-test: | kill build ## Run basic integration tests after npm install in a
                CMD=$(CMD) \
                SERVER_NAME=$(IT_SERVER_NAME) \
                DOCKER_OPTIONS="-e APP_URL=http://$(DASHBOARD_CONTAINER_NAME):$(HTTP_PORT) \
-                       -e KERNEL_GATEWAY_URL=http://$(KG_CONTAINER_NAME):$(IT_KG_PORT) \
+                       -e KERNEL_GATEWAY_URL=http://$(KG_CONTAINER_NAME):$(KG_PORT) \
                        -e KG_AUTH_TOKEN=$(KG_AUTH_TOKEN) \
                        -e AUTH_TOKEN=$(AUTH_TOKEN) \
                        -e TEST_USERNAME=$(USERNAME) \
```

**Comment by parente** [comment]
Using an overridable var for `KG_PORT?=` is wrong in the case of the linked containers because they actually communicate on the internal docker port which is always 8888. I think we're all over the map on how we're doing ports here. I'll try to tidy it up.

File: Makefile (hunk 1 above)
Comment (parente) [inline Makefile]: Fixed in the commit below.

**Comment by jhpedemonte** [comment]
Looks good now. Works for me on Docker beta and docker-machine.

**Pull request merged**